### PR TITLE
Rename TooOldPythonError to UnsupportedPythonError

### PR DESCRIPTION
**Pull request by astrofrog**
TooOldPythonError sounds a bit too colloquial

EDIT: Follow up of #6595

### CODE REVIEW

File: astropy/__init__.py [hunk 1]
```diff
@@ -15,9 +15,11 @@
 __minimum_numpy_version__ = '1.10.0'
 
 
-class TooOldPythonError(Exception): pass
+class UnsupportedPythonError(Exception):
+    pass
```
Comment (eteq) [inline astropy/__init__.py]: This is super-nitpicky, but I think these content-less exceptions are better as a one-liner.

Comment (astrofrog) [reply astropy/__init__.py]: Does PEP8 allow one-liners? I really don't like the one liners...

Comment (bsipocz) [reply astropy/__init__.py]: I've just checked the same thing. People argue it does: https://github.com/PyCQA/pycodestyle/issues/231#issuecomment-185021750, but the tools doesn't recognize them yet.

Comment (pllim) [reply astropy/__init__.py]: Off topic: @eteq , just the other day, you cringed when I said one could write an entire C function in one line...

p.s. FWIW two lines here looks cleaner to me.

**Pull request merged**